src/psyclone/gocean1p0.py

```diff
@@ -833,8 +876,11 @@ def gen_code(self, parent):
 
         # Generate the upper and lower loop bounds
         # TODO: Issue 440. upper/lower_bound should generate PSyIR
-        self.start_expr = Literal(self._lower_bound(), parent=self)
-        self.stop_expr = Literal(self._upper_bound(), parent=self)
+        self.start_expr = self._lower_bound(parent=self)
+        self.stop_expr = self._upper_bound(parent=self)
+        # Update the loop limits for the parent loop
+        self.children[0] = self.start_expr
```

Comment:
sn't self.start_expr the same as self.children[0]? Similarly for self.stop_expr? If so, you don't need these lines.

Reply:
As above.